tests/data_out/data_out_postprocessor_vector_complex_01.cc

```diff
@@ -0,0 +1,206 @@
+#include <deal.II/numerics/vector_tools.h>
+
+#include "../tests.h"
+
+
+
+std::ofstream logfile("output");
+
+
+template <int dim>
+class LaplaceProblem
+{
+public:
+  LaplaceProblem();
+  void
+  run();
+
+private:
+  void
+  make_grid_and_dofs();
+  void
+  solve();
+  void
+  output_results() const;
+
+  Triangulation<dim> triangulation;
+  FESystem<dim>      fe;
+  DoFHandler<dim>    dof_handler;
+
+  Vector<std::complex<double>> solution;
+};
+
+
+template <int dim>
+LaplaceProblem<dim>::LaplaceProblem()
+  : fe(FE_Q<dim>(1), 2)
+  , dof_handler(triangulation)
+{}
+
+
+
+template <int dim>
+void
+LaplaceProblem<dim>::make_grid_and_dofs()
+{
+  GridGenerator::hyper_cube(triangulation, 0, 1);
+  triangulation.refine_global(1);
+  triangulation.begin_active()->set_refine_flag();
+  triangulation.execute_coarsening_and_refinement();
+
+  dof_handler.distribute_dofs(fe);
+  solution.reinit(dof_handler.n_dofs());
+}
+
+
+
+template <int dim>
+class SinesAndCosines : public Function<dim, std::complex<double>>
+{
+public:
+  SinesAndCosines()
+    : Function<dim, std::complex<double>>(2)
+  {}
+
+  std::complex<double>
+  value(const Point<dim> &p, const unsigned int component) const
+  {
+    switch (component)
+      {
+        case 0:
+          return std::sin(p.norm());
+        case 1:
+          return std::cos(p.norm());
+        default:
+          Assert(false, ExcNotImplemented());
+          return 0;
+      }
+  }
+};
+
+
+
+template <int dim>
+void
+LaplaceProblem<dim>::solve()
+{
+  // dummy solve. just insert some
+  // values as mentioned at the top
+  // of the file
+  VectorTools::interpolate(dof_handler, SinesAndCosines<dim>(), solution);
+}
+
+
+template <int dim>
+class MyPostprocessor : public DataPostprocessorVector<dim>
+{
+public:
+  MyPostprocessor()
+    : DataPostprocessorVector<dim>("magnitude_times_d", update_values)
+  {}
+
+  virtual void
+  evaluate_vector_field(const DataPostprocessorInputs::Vector<dim> &input_data,
+                        std::vector<Vector<double>> &computed_quantities) const
+  {
+    for (unsigned int q = 0; q < input_data.solution_values.size(); ++q)
+      {
+        Assert(computed_quantities[q].size() == dim, ExcInternalError());
+
+        for (unsigned int d = 0; d < dim; ++d)
+          computed_quantities[q](d) =
+            input_data.solution_values[q](0) *
+              input_data.solution_values[q](0) +
+            input_data.solution_values[q](1) * input_data.solution_values[q](1);
+        AssertThrow(std::fabs(computed_quantities[q](0) - 1) < 1e-12,
+                    ExcInternalError());
+      }
+  }
+};
+
+
+
+template <int dim>
+void
+LaplaceProblem<dim>::output_results() const
+{
+  {
+    // Save the data in the log file
+    MyPostprocessor<dim> p;
+    DataOut<dim>         data_out;
+    data_out.attach_dof_handler(dof_handler);
+    data_out.add_data_vector(solution, p);
+    data_out.build_patches();
+    data_out.write_gnuplot(logfile);
```

Comment:
I run this test and output the data  as `solution.vtu`. There is only one output which is `magnitudes_times_d`. This output has 3 components for x, y and z. There are no real and imaginary parts. Is this the expected behavior?

I used:
``` c++
std::ofstream output("solution.vtu");
data_out.write_vtu(output);
```

Reply:
That's what you specify in the test: That it's a `DataPostprocessorVector` (which has `dim` output components), and you provide the name "magnitude_times_d" to this output vector. So yes, by virtue of deriving from this base class, you say that the output is a real-valued vector.